### ISSUE TITLE
conan install causes conanbuildinfo.cmake to be re-generated even if it hasn't changed

### ISSUE THREAD

**Issue by avf** [conan-io/conan#2895]
I've written my own buildscript that attempts to build my entire project including all dependencies in a single step.

On every build, my script calls `conan install` to install the dependencies and then runs `cmake` and `ninja` to build my own source files. I'm calling `conan install` before every build, because otherwise the build process won't pick up any changes in the consumer `conanfile.py` (for example a version update of a dependency). Unfortunately, every time I call `conan install`, conan will re-generate the `conanbuildinfo.cmake`, even when its contents haven't changed. Now the cmake/ninja step will cause a full rebuild of my own sources, because the timestamp on `conanbuildinfo.cmake` was updated. This rebuild is unnecessary (and takes a long time), since nothing has actually changed. Wouldn't it be better if conan only generated that file if it doesn't yet exist or if any changes were made? Or am I missing something?

I'm running conan 1.3.3 on macOS and my build system is CMake+ninja.

**Comment by memsharded** [comment]
Sounds a reasonable request, and shouldn't be difficult to implement, lets have a look.

**Comment by avf** [comment]
Awesome, thanks for your quick response!

For anyone else encountering this issue, as a temporary workaround, I've modified my buildscript to do the following:

- Before calling `conan install`, I check if `conanbuildinfo.cmake` exists. If so, I read its contents into memory and rename the file. Then I call `conan install` and read the newly generated `conanbuildinfo.cmake`. If the contents of the new and the old file matches, I delete the new file and restore the old file. Otherwise I delete the old file.

Here's the code:

```(python)
	conan_buildinfo_path = build_dir + "/conanbuildinfo.cmake"
	old_conan_buildinfo_path = build_dir + "/old_conanbuildinfo.cmake"
	conan_buildinfo_exists = os.path.isfile(conan_buildinfo_path)
	old_conan_buildinfo_data = ""
	if conan_buildinfo_exists:
		with open(conan_buildinfo_path, 'r') as conan_buildinfo_file:
			old_conan_buildinfo_data = conan_buildinfo_file.read()
		os.rename(conan_buildinfo_path, old_conan_buildinfo_path)

        # Call conan install

        if old_conan_buildinfo_data != "":
		with open(conan_buildinfo_path, 'r') as conan_buildinfo_file:
			new_conan_buildinfo_data = conan_buildinfo_file.read()
			if old_conan_buildinfo_data == new_conan_buildinfo_data:
				os.remove(conan_buildinfo_path)
				os.rename(old_conan_buildinfo_path, conan_buildinfo_path)
			else:
				os.remove(old_conan_buildinfo_path)

```

**Comment by memsharded** [comment]
Done, will be released in conan 1.8
